exp/addons/internal/controllers/clusterresourceset_controller.go

```diff
@@ -264,6 +264,9 @@ func (r *ClusterResourceSetReconciler) ApplyClusterResourceSet(ctx context.Conte
 		}
 	}()
 
+	// Ensure that the owner references are set on the ClusterResourceSetBinding.
+	clusterResourceSetBinding.OwnerReferences = ensureOwnerRefs(clusterResourceSetBinding, clusterResourceSet, cluster)
```

Comment:
Note this is done in the ClusterResourceSet reconcile rather than the ClusterResourceSetBinding reconcile. This may cause some slight drift in setting the owner references. 

The main case we care about is backup -> restore -> deletion so the drift of a couple of minutes should be acceptable. In the normal case we'd expect these owner references to be applied on the first reconcile of the ClusterResourceSet i.e. right after restore + unpause.